src/nbody_simulation.jl

```diff
@@ -0,0 +1,51 @@
+using StaticArrays, DiffEqBase, OrdinaryDiffEq, RecipesBase, DiffEqCallbacks
+
+include("./nbody_bodies.jl")
+include("./nbody_basic_potentials.jl")
+include("./nbody_system.jl")
+include("./nbody_boundary_conditions.jl")
+
+
+# This structure defines conditions under wich we test our system of n-bodies
+# With this wrapping we can make such fields as `boundary_conditions` necessary for every simulation
+# while allowing one to describe a particular systme of N interacting particles
+struct NBodySimulation{sType <: NBodySystem,bcType <: BoundaryConditions}
+    system::sType
+    tspan::Tuple{Float64,Float64}
+    boundary_conditions::bcType
+    external_electric_field
```

Comment:
the way that there are applied doesn't cause performance issues? If it's just a single modifying call then it's fine. From what I can see it looks fine?

Reply:
Right now it isn't implemented: I left it here just to show where I want to set these fields up. In the future, I think, these fields will depend on time and coordinates